rawrequest/client_test.go

```diff
@@ -0,0 +1,130 @@
+package rawrequest
+
+import (
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	assert "github.com/stretchr/testify/require"
+	stripe "github.com/stripe/stripe-go/v79"
+	_ "github.com/stripe/stripe-go/v79/testing"
+)
+
+func stubAPIBackend(testServer *httptest.Server) {
+	backend := stripe.GetBackendWithConfig(
+		stripe.APIBackend,
+		&stripe.BackendConfig{
+			LeveledLogger: &stripe.LeveledLogger{
+				Level: stripe.LevelDebug,
+			},
+			MaxNetworkRetries: stripe.Int64(0),
+			URL:               stripe.String(testServer.URL),
+		},
+	).(*stripe.BackendImplementation)
+
+	stripe.SetBackend(stripe.APIBackend, backend)
+}
+
+func TestV2PostRequest(t *testing.T) {
+	var body string
+	var path string
+	var method string
+	var contentType string
+	testServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		req, _ := ioutil.ReadAll(r.Body)
+		r.Body.Close()
+		body = string(req)
+		path = r.URL.RequestURI()
+		method = r.Method
+		contentType = r.Header.Get("Content-Type")
```

Comment:
may want to assert on contentType == application/json, or return a 400 if its not, or something

Reply:
we do farther down - this line stores the incoming value on the "server" and then we assert its value after the response finishes.

Reply:
https://github.com/stripe/stripe-go/pull/1924/files#diff-d25e28f49be803aa8fb79967e7cacada37a257743d0d82e8c4585b5a26ab9e1fR56

Reply:
ah i see that now.  thanks!